contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_hexadecimal.py

```diff
@@ -1,75 +1,41 @@
-# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
-
-
-class ColumnValuesToBeHexadecimal(ColumnMapMetricProvider):
-    # This is the id string that will be used to reference your metric.
-    condition_metric_name = "column_values.is_hexadecimal"
-
-    filter_column_isnull = False
-
-    # This method implements the core logic for the PandasExecutionEngine
-    @column_condition_partial(engine=PandasExecutionEngine)
-    def _pandas(cls, column, **kwargs):
-        def is_hex(x):
-            if not x:
-                return False
-            if x is None:
-                return False
-            if not isinstance(x, str):
-                return False
-            try:
-                int(x, 16)
-                return True
-            except ValueError:
-                return False
 
-        return column.apply(is_hex)
-
-    # This method defines the business logic for evaluating your metric when using a SqlAlchemyExecutionEngine
-    # @column_condition_partial(engine=SqlAlchemyExecutionEngine)
-    # def _sqlalchemy(cls, column, _dialect, **kwargs):
-    #     raise NotImplementedError
+class ExpectColumnValuesToBeHexadecimal(RegexBasedColumnMapExpectation):
+    """Expect column values to be valid hexadecimals."""
 
-    # This method defines the business logic for evaluating your metric when using a SparkDFExecutionEngine
-    # @column_condition_partial(engine=SparkDFExecutionEngine)
-    # def _spark(cls, column, **kwargs):
-    #     raise NotImplementedError
+    regex_camel_name = "HexadecimalNumber"
+    regex = "^[0-9a-fA-F]+$"
+    semantic_type_name_plural = "hexadecimals"
 
+    map_metric = RegexBasedColumnMapExpectation.register_metric(
+        regex_camel_name=regex_camel_name,
+        regex_=regex,
+    )
 
-# This class defines the Expectation itself
-class ExpectColumnValuesToBeHexadecimal(ColumnMapExpectation):
-    """Expect column values to be valid hexadecimals."""
+    library_metadata = {
+        "maturity": "experimental",
+        "tags": ["experimental"],
+        "contributors": [
+            "@andrewsx",
+            "@mkopec87",
+        ],
+    }
 
-    # These examples will be shown in the public gallery.
-    # They will also be executed as unit tests for your Expectation.
     examples = [
         {
             "data": {
                 "a": ["3", "aa", "ba", "5A", "60F", "Gh"],
                 "b": ["Verify", "String", "3Z", "X", "yy", "sun"],
                 "c": ["0", "BB", "21D", "ca", "20", "1521D"],
                 "d": ["c8", "ffB", "11x", "apple", "ran", "woven"],
-                "e": ["a8", "21", 2.0, "1B", "4AA", "31"],
-                "f": ["a8", "41", "ca", 46, "4AA", "31"],
-                "g": ["a8", "41", "ca", "", "0", "31"],
-                "h": ["a8", "41", "ca", None, "0", "31"],
+                "e": ["a8", "21", "2.0", "1B", "4AA", "31"],
+                "f": ["a8", "41", "ca", "", "0", "31"],
```

Comment:
After line 39, can you add `"suppress_test_for": ["bigquery", "snowflake"],`

Reply:
Done. I have a question though: should I setup local connections for bigquery and snowflake to test compatibility with those backends? How can I test them? Or they should be suppresed for some other reason?

Reply:
Oh no that's not anything you need to do yourself. The CI/CD process does that and I happen to have creds for our test snowflake/redshift/bigquery locally